cmake/Dependencies.cmake

```
@@ -1018,9 +1018,30 @@ if(USE_MPI)
       if(_output MATCHES "smcuda")
         message(STATUS "Found OpenMPI with CUDA support built.")
       else()
-        message(WARNING "OpenMPI found, but it is not built with CUDA support.")
-        set(CAFFE2_FORCE_FALLBACK_CUDA_MPI 1)
+        if(USE_CUDA_MPI)
+            message(WARNING "OpenMPI with CUDA support not found, but forcing anyway.")
+            set(CAFFE2_USE_CUDA_MPI 1)
+        else()
+            message(WARNING "OpenMPI found, but it is not built with CUDA support.")
+            set(CAFFE2_FORCE_FALLBACK_CUDA_MPI 1)
+        endif()
       endif()
+    else()
```

Please consider adding comment inside `else()` statement to better track nested if-else-endif statements
```suggestion
    else(MPI_CXX_FOUND)
```